test/gtest/ucp/ucp_test.cc

```diff
@@ -476,8 +476,9 @@ ucs_status_t ucp_test_base::entity::listen(const struct sockaddr* saddr,
     restore_errors();
     if (status == UCS_OK) {
         m_listener.reset(listener, ucp_listener_destroy);
-    } else if (status != UCS_ERR_INVALID_ADDR) {
-        /* throw error if status is not (UCS_OK or UCS_ERR_INVALID_ADDR) */
+    } else if (status != UCS_ERR_UNREACHABLE) {
+        /* throw error if status is not (UCS_OK or UCS_ERR_UNREACHABLE).
+         * UCS_ERR_INVALID_PARAM may also return but then the test should fail */
         ASSERT_UCS_OK(status);
```

Comment:
`OK` is handled above, it's always `false`. also, is it really a reason to abort the testing? maybe replace `assert` with `expect`?

Reply:
right, this assert is only to make it throw an error in this case.

Reply:
Then it can be simplified and more clear:
```
if (status == UCS_OK) {
    m_listener.reset(listener, ucp_listener_destroy);
} else {
    ASSERT_EQ(UCS_ERR_UNREACHABLE, status);
}